### ISSUE TITLE
Applying a "style / preset" results in multiple undoable steps

### ISSUE THREAD

**Issue by dlemmermann** [gluonhq/rich-text-area#199]
Applying a style from a list of presets can sometimes only be done by applying more than one decoration. For example: switching to or from a numbered list requires the adding / removal of the graphic decoration in addition to setting the correct font family name and font size. In our application we have added an `apply()` method to the `Presets` enumerator and it now looks like this:

```java
        public void apply(RichTextArea area) {
            area.getActionFactory().decorate(TextDecoration.builder()
                            .fontSize(getFontSize())
                            .fontWeight(getFontWeight())
                            .fontFamily(getFontFamily())
                            .build())
                    .execute(new ActionEvent());

            if (isBulletList()) {
                area.getActionFactory().decorate(ParagraphDecoration.builder()
                                .graphicType(ParagraphDecoration.GraphicType.BULLETED_LIST)
                                .build())
                        .execute(new ActionEvent());
            } else if (isNumberList()) {
                area.getActionFactory().decorate(ParagraphDecoration.builder()
                                .graphicType(ParagraphDecoration.GraphicType.NUMBERED_LIST.NUMBERED_LIST)
                                .build())
                        .execute(new ActionEvent());
            } else {
                area.getActionFactory().decorate(ParagraphDecoration.builder()
                                .graphicType(ParagraphDecoration.GraphicType.NONE)
                                .build())
                        .execute(new ActionEvent());
            }
        }
```

With this we end up having two commands on the undo stack and the operation can only be undone by pressing the undo button several times. We either need a command that combines multiple sub-commands or we need the higher-level concept of a "Style / Preset" directly inside the framework.

**Comment by dlemmermann** [comment]
I think what is really needed is 1st class support for presets directly in the API. A preset can result in multiple decorations being applied. A preset is applied as a single command (maybe a composite command). A preset can be undone / redone in a single step. @sten-navie your thoughts?

**Comment by sten-navie** [comment]
My thought is that I trust you @dlemmermann with your assesment here. The app that Dirk works on, is one where we do not want the user to select fonts and their sizes completely freely, so we are doing presets like "Title", "Heading", "Sub Heading". "Text Body". So this is a functionality that we want in some way or other.

If it is implemented via 1st class support for presets or by some other mechanism where we can group multiple changes into a single undoable change, is not so important for me as long as we functionally are able to provide presets.

@erwin1 and @jperedadnr  I assume that you also might have some opinions on how this would be best implemented. This was not something that we had in the original spec, so it is not a MVP-phase thing, but it is definitely important to us.

**Comment by jperedadnr** [comment]
This will be fixed with #192
